### PR TITLE
Enable TestBed in nativescript

### LINKED ISSUE #478: Using TestBed with NativeScript
How can I use TestBed with NativeScript. This spec is not working:

```
import 'reflect-metadata';
import {Router} from '@angular/router';
import {BaseRequestOptions, Response, Http, ResponseOptions} from '@angular/http';
import {MockBackend} from '@angular/http/testing';
import {Page} from 'ui/page';
import {BrowserDynamicTestingModule, platformBrowserDynamicTesting} from '@angular/platform-browser-dynamic/testing';
import {inject, TestBed} from '@angular/core/testing';
import {SettingsService, AuthService, Logger, User} from '@abc/angular-common';
import * as appSettings from 'application-settings';
import {ApplicationSettingsService} from '../../services/app_settings.service';
import {AppLogService} from '../../services/app_logger.service';
import {RouterMock} from '../mocks/router.mock';
import {ProfileComponent} from '../../components/profile/profile.component';
import {MockConnection} from '@angular/http/testing/mock_backend';

TestBed.initTestEnvironment(BrowserDynamicTestingModule, platformBrowserDynamicTesting());

describe('ProfileComponent', () => {

  // provide our implementations or mocks to the dependency injector
  beforeEach(() => {
    appSettings.clear();
    TestBed.configureTestingModule({
      providers: [
        {provide: Router, useClass: RouterMock},
        {provide: Logger, useClass: AppLogService},
        {provide: SettingsService, useClass: ApplicationSettingsService},
        BaseRequestOptions,
        MockBackend,
        {
          provide: Http,
          useFactory: (backend, defaultOptions) => {
            return new Http(backend, defaultOptions);
          },
          deps: [MockBackend, BaseRequestOptions]
        },
        AuthService,
        Page,
        ProfileComponent
      ]
    });
  });

  it('should init', inject([ProfileComponent], (profile) => {
    expect(profile).toBeDefined();
  }));

  it('should get user on init', inject([ProfileComponent, AuthService], (profile, authService) => {
    let user: User = new User();
    user.username = 'test';

    spyOn(authService, 'getUser').and.callFake(() => {
      return user;
    });

    // trigger init to get user
    profile.ngOnInit();
    expect(authService.getUser).toHaveBeenCalled();
    expect(profile.user).toBe(user);
  }));

});

```

TestBed would be nice as it gives the ability to have several injects in a test case

### PR DESCRIPTION
**Pull request by hypery2k**
Fixes #478, #479 and #584

Currently having issues that the setup for TestBed is working, but other tests are failing:
```
START:
  ✖ "before each" hook for "creates component"

Finished in 0.091 secs / 0.001 secs
```

If I use it standalone all is working fine. Should I add a new test app for that?

### CODE REVIEW

**Comment by hdeshev** [comment]
This looks like it's related to #733 , yet it doesn't seem to touch the setGlobalVar implementation, so I'll assume it doesn't fix that issue.

File: nativescript-angular/testing/setup.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require("reflect-metadata");
+require("nativescript-angular/zone-js/dist/zone-nativescript");
+require("zone.js/dist/long-stack-trace-zone");
```
Comment (hdeshev) [inline nativescript-angular/testing/setup.js]: NativeScript bundles its own version of zone.js, and requiring `zone.js` from the npm looks fishy. You could easily end up with two separate zone implementations and that will likely break things. I suggest we use `nativescript-angular/zone-js` everywhere.

File: nativescript-angular/testing/setup.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+require("zone.js/dist/proxy"); // since zone.js 0.6.15
+require("zone.js/dist/sync-test");
+if (global.jasmine) {
+    require("zone.js/dist/jasmine-patch"); // put here since zone.js 0.6.14
+} 
+require("zone.js/dist/async-test");
+require("zone.js/dist/fake-async-test");
+var testing_1 = require("@angular/platform-browser-dynamic/testing");
```
Comment (hdeshev) [inline nativescript-angular/testing/setup.js]: This looks like code transpiled from TypeScript. Why not implement it in TypeScript instead?

File: nativescript-angular/testing/setup.js (hunk 1 above)
Comment (Rachnerd) [inline nativescript-angular/testing/setup.js]: Does this mean that you also want to include the jasmine-patch + mocha-patch for unit testing? Those are currently necessary to run our tests and they're not included in nativescript-angular/zone-js.

Comment (hypery2k) [inline nativescript-angular/testing/setup.js]: @hdeshev should i create a PR at nativescript-angular/zone-js?

File: nativescript-angular/testing/setup.js (hunk 2 above)
Comment (hypery2k) [inline nativescript-angular/testing/setup.js]: i started with TypeScript, but had some issues with module dependencies, but will check it again

**Comment by hypery2k** [comment]
I think I will rebase my fork and make a new PR, or?

**Comment by sis0k0** [comment]
Yes, that will be good, thank you :)

**Comment by hypery2k** [comment]
see #751 and #752

**Pull request closed without merging**